CRM/Core/DAO.php

```diff
@@ -2283,39 +2283,25 @@ public static function checkTriggerViewPermission($view = TRUE, $trigger = TRUE)
     }
     // test for create view and trigger permissions and if allowed, add the option to go multilingual
     // and logging
-    // I'm not sure why we use the getStaticProperty for an error, rather than checking for DB_Error
-    CRM_Core_TemporaryErrorScope::ignoreException();
-    $dao = new CRM_Core_DAO();
-    if ($view) {
-      $result = $dao->query('CREATE OR REPLACE VIEW civicrm_domain_view AS SELECT * FROM civicrm_domain');
-      if (PEAR::getStaticProperty('DB_DataObject', 'lastError') || is_a($result, 'DB_Error')) {
-        return FALSE;
+    try {
+      if ($view) {
+        CRM_Core_DAO::executeQuery('CREATE OR REPLACE VIEW civicrm_domain_view AS SELECT * FROM civicrm_domain');
       }
-    }
 
-    if ($trigger) {
-      $result = $dao->query('CREATE TRIGGER civicrm_domain_trigger BEFORE INSERT ON civicrm_domain FOR EACH ROW BEGIN END');
-      if (PEAR::getStaticProperty('DB_DataObject', 'lastError') || is_a($result, 'DB_Error')) {
-        if ($view) {
-          $dao->query('DROP VIEW IF EXISTS civicrm_domain_view');
-        }
-        return FALSE;
+      if ($trigger) {
+        CRM_Core_DAO::executeQuery('CREATE TRIGGER civicrm_domain_trigger BEFORE INSERT ON civicrm_domain FOR EACH ROW BEGIN END');
+        CRM_Core_DAO::executeQuery('DROP TRIGGER IF EXISTS civicrm_domain_trigger');
       }
 
-      $dao->query('DROP TRIGGER IF EXISTS civicrm_domain_trigger');
-      if (PEAR::getStaticProperty('DB_DataObject', 'lastError')) {
-        if ($view) {
-          $dao->query('DROP VIEW IF EXISTS civicrm_domain_view');
-        }
-        return FALSE;
+      if ($view) {
+        CRM_Core_DAO::executeQuery('DROP VIEW IF EXISTS civicrm_domain_view');
       }
     }
-
-    if ($view) {
-      $dao->query('DROP VIEW IF EXISTS civicrm_domain_view');
-      if (PEAR::getStaticProperty('DB_DataObject', 'lastError')) {
-        return FALSE;
+    catch (Exception $e) {
+      if ($view) {
+        CRM_Core_DAO::executeQuery('DROP VIEW IF EXISTS civicrm_domain_view');
```

Comment:
If the create / replace failed as per old line 2290 we would have returned before trying to drop the view right?

Reply:
erm - I'm confused what 2290 is - it got removed - have you checked this out or are you looking at it on github?

Reply:
Where would it be returning other than the line after this?

Reply:
looking on github using the ?w=1 added to the url

Reply:
so if the DB caused an error as per `if (PEAR::getStaticProperty('DB_DataObject', 'lastError') || is_a($result, 'DB_Error')) {` it would return FALSE before it get to the point of dropping the view it would trigger a return FALSE right? and Also we aren't using Exceptions right?

Reply:
@seamuslee001 we should be using exceptions! The code to suppress them was ```CRM_Core_TemporaryErrorScope::ignoreException();``` which would not work because it was instantly deconstructed. Any path to get to this code should be with-exceptions

Reply:
@eileenmcnaughton my point is that just removing that doesn't cause executeQuery to throw exceptions AFAIK but more importantly I don't think the revised code matches what would happen if there was a DB error running the create or replace query at least

Reply:
@seamuslee001 as long as it's called from the api or a form it should through exceptions - if it IS throwing exceptions then it should be the same - ie catch the exception, drop the view, return false

Reply:
@eileenmcnaughton so in the previous incarnation it wouldn't drop the view if it failed because this code would have caused it to return FALSE before it got to the drop view query

```
    CRM_Core_TemporaryErrorScope::ignoreException();
    $dao = new CRM_Core_DAO();
    if ($view) {
      $result = $dao->query('CREATE OR REPLACE VIEW civicrm_domain_view AS SELECT * FROM civicrm_domain');
      if (PEAR::getStaticProperty('DB_DataObject', 'lastError') || is_a($result, 'DB_Error')) {
        return FALSE;
      }
    }
```

Reply:
@seamuslee001 so it only drops the view if exists - but we could put an extra try-catch around that in case it fails!